### DISCUSSION TITLE
Display issue? Consuming finished, but still "processing" and "waiting" according to Dashboard

### DISCUSSION THREAD

**Discussion by TheNomad11** [jonaswinkler/paperless-ng#724] [General]
Everytime after I have uploaded a document through the web interface, the status message in the dashboard says "processing", displaying the recently uploaded filenames below with the messages "Upload compete, waiting". But everything went fine, and in the logs there are no error messages either. Maybe just a display issue? After reloading the page, these messages disappear

**Comment by jonaswinkler** [comment]
Further status messages after "Upload complete, waiting" are provided via WebSockets. 

* With the docker image, this should work out of the box when directly connecting to paperless (no proxy webserver). If this is what you're doing, then this is a bug.
* With a proxy server in front of paperless, you need some additional configuration for websockets to work. The documentation has some instructions for nginx, and apache should be similar.
* If you're using Apache `mod_wsgi`, websockets won't work at all.

You can press `F12` and look for any errors in the console or networking tab to find out if WebSockets are not working.

**Reply by TheNomad11** [reply]
Aha ok, thanks for quick answer. I used the docker image and then a (too?) barebones Apache reverse proxy. Should I add the websockets part or won't it work with Apache? I am new to reverse proxy and still learning:

```
<IfModule mod_ssl.c>
<VirtualHost *:443>
    ServerName paperless.xxxxx.com
    AllowEncodedSlashes On
    ProxyPreserveHost On
    ProxyPass "/"  "http://localhost:8888/"

SSLCertificateFile /etc/letsencrypt/live/XXXXXX/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/xxxxxxx/privkey.pem
Include /etc/letsencrypt/options-ssl-apache.conf
</VirtualHost>
</IfModule>
```

**Reply by jonaswinkler** [reply]
You'll most certainly need [mod_proxy_wstunnel](https://httpd.apache.org/docs/2.4/mod/mod_proxy_wstunnel.html).

It might be enough to just enable that module. If not, additional configuration is required. I don't have a setup ready right now, so I can't provide any details. WebSocket connections are served at `http://localhost:8888/ws/...` and need to be forwarded to `https://<public-host>/ws/...`.

Maybe this will work:

```
ProxyPassMatch ^/(ws)$  ws://localhost:8888/$1
ProxyPass / http://localhost:8888/
```

If you find a minimal configuration that works, I'd appreciate a PR for the documentation below https://paperless-ng.readthedocs.io/en/latest/setup.html#using-nginx-as-a-reverse-proxy !

**Reply by TheNomad11** [reply]
Thanks. I played a bit with different combinations, but it is still stuck with waiting. But that is ok for me as the document is processed correctly, it seems. Maybe someone with more Apache knowledge has to step in here. Generally, the majority seems to prefer Nginx, though
